counter-sink/src/main/resources/application.yml

```diff
@@ -5,4 +5,4 @@ spring:
   cloud:
     stream:
       bindings:
-        input: counter
+        input: ${remote.name:application:${server.port}}
```

Comment:
The idea to introduce `application:${server.port}` as the default value for the binding name was to use the same context Id spring boot [sets] (https://github.com/spring-projects/spring-boot/blob/master/spring-boot/src/main/java/org/springframework/boot/context/ContextIdApplicationContextInitializer.java#L60). Since we use embedded servlet container via spring boot web starter, the `server.port` will always be set.